policy/release/base_image_registries.rego

```diff
@@ -24,15 +24,10 @@ import data.lib.bundles
 # custom:
 #   short_name: disallowed_base_image
 #   failure_msg: Base image %q is from a disallowed registry
-#   rule_data:
-#     allowed_registry_prefixes:
-#     - registry.access.redhat.com/
-#     - registry.redhat.io/
 #
 deny contains result if {
-	allowed_registry_prefixes := lib.rule_data(rego.metadata.rule(), "allowed_registry_prefixes")
```

Comment:
One idea:

We could add some logic to lib.rule_data so that it first looks for `data.values.allowed_registry_prefixes` and if that is not found then it falls back to the default rule_data defined in the annotations.

It mean the rule could be functional out of the box without any additional data, with a clear path to providing the custom data when it's required.

Reply:
My concern with providing default values is that it creates two paths for resolving data values. Since we control the policies repo, we'll have the motivation to always just use the default values for our purposes. This leaves room for the other mechanism to break over time without anyone actually noticing until someone other than us tries to use it.

Reply:
I think that is a valid concern and that it can be mitigated with a test that removes the annotations and relies on the data files alone. So with that I would have the default values in the annotations and a way to override them via data files.

Reply:
Fair point. We could do that with unit tests without much effort.

Reply:
Yeah I'm not 100% sold on the "fallback to annotations" idea, but I guess it would make it easy to use one or the other depending on what works best for the particular rule.

Imagine a list of weekdays for example, might as well ship them with the rule definition. There's likely no need to override them in a user created data source.

Reply:
I'm in the minority here, but I think adding support for fallback adds little value. Today, we already need to provide data when evaluating the rules because some of the data we use is just too large to fit in the annotations, i.e. acceptable bundles.

By introducing a fallback mechanism, we basically have to double the amount of tests for policies that use data. I think that's unnecessary toil to support a feature we'll likely never benefit from.

Reply:
To be clear, I'm in favor of ditching the fallback idea of "look here, if not found look there", especially since the rego.metadata.chain() always needs to be inside the rule, which I guess means it would always have to be there in the function call.

Actually I suppose individual rules could define their own fallback mechanism if it was considered useful, but I think it's best not to make that the convention for everything.